File: Decorator/WeaponSample/src/ExtendedMag.java
public class ExtendedMag extends WeaponAttachment {
    public ExtendedMag(IWeapon weapon) {
        super(weapon);
        this.name = "Extended Mag";
    }

    @Override
    public void Fire() {
        weapon.Fire();
        System.out.println("    " + this.name);
    }
}

File: Decorator/csharp/Sample/Decorator/Weapon/ExtendedMag.cs
using System;

namespace Decorator.Weapon
{
    public class ExtendedMag : WeaponAttachment
    {
        public ExtendedMag(IWeapon weapon) : base(weapon)
        {
        }
        public override void Fire()
        {
            _weapon.Fire();
            Console.WriteLine("     Extended Mag");

        }
    }
}